overlord/devicestate/systems.go

```diff
@@ -176,3 +186,212 @@ func seededSystemFromModeenv() (*seededSystem, error) {
 	}
 	return seededSys, nil
 }
+
+func createSystemForModelFromValidatedSnaps(getInfo func(name string) (*snap.Info, bool, error), db asserts.RODatabase, label string, model *asserts.Model) (newFiles []string, dir string, err error) {
+	if isUC20 := model.Grade() != asserts.ModelGradeUnset; !isUC20 {
+		return nil, "", fmt.Errorf("cannot create a system for non UC20 model")
+	}
+
+	logger.Noticef("creating recovery system with label %q for %q", label, model.Model())
+
+	// TODO: should that path provided by boot package instead?
+	recoverySystemDirInRootDir := filepath.Join("/systems", label)
+	assertedSnapsDir := filepath.Join(boot.InitramfsUbuntuSeedDir, "snaps")
+	recoverySystemDir := filepath.Join(boot.InitramfsUbuntuSeedDir, recoverySystemDirInRootDir)
+
+	wOpts := &seedwriter.Options{
```

Comment:
how much of this logic really needs to live inside devicestate? I guess I just am reading through this code and having to jump back and forth in the seedwriter package to ensure that all the required steps are being done in the right order and such, and I don't wonder if this wouldn't be simpler to read and more clear if the big chunks of this lived in seedwriter alongside all the other complicated logic in that package?